mantidproject/mantid
# Indirect ILL : fix issue with crop dead monitor channels

This PR fixes 2 minor issues in indirect ILL reduction encountered in a pathological cases as explained below:

**To test:**
Set ILL/IN16B as facility/instrument.

1) IN16B can record the data in mirror mode, when the raw data is organised in two adjacent equal wings, as shown, where each wing (left or right half of the matrix) has e.g. 6 bins.

![screen shot 2017-06-12 at 13 56 57](https://user-images.githubusercontent.com/16865759/27032851-04d17bea-4f77-11e7-9ace-489e2e16ac9d.png)

In the reduction there is an option to crop out the bins, where the monitor (the first row) has 0 counts.
In the common case such bins are the same for the left and the right wings and there is no trouble.

However in a pathological case, as shown in the figure, 0 monitor bins are not exactly the same (0,1,6 for the left, 0,5,6 for the right) , so after cropping one is left with different axes, which would be later impossible to sum. The solution is to shift the x-axis after cropping to start from zero, which is done in https://github.com/mantidproject/mantid/pull/19846/files#diff-6cdb203e68b9cc4f584faeece4167d87R338

In a yet other imaginary case, it can be that the number of bins with 0 count is different in left and right. In this case nothing can be done rather than raising an error with a comprehensive message (https://github.com/mantidproject/mantid/pull/19846/files#diff-ee33a12732aafd5eceda9c75ff15a864R445) to prevent Plus from failing later.

2) Another issue has been fixed in `IndirectILLReductionFWS` regarding the calibration scale when interpolation option is requested.



**To test**

**1)** 
```qens = IndirectILLReductionQENS(Run='140721-140722', CropDeadMonitorChannels=True) ``` 
Should run without errors. A new test is added for this problematic case.

**2)** 
```
fws_sum = IndirectILLReductionFWS(Run='141195:141200', CalibrationRun='140645:140650', CalibrationOption='Sum')
fws_int = IndirectILLReductionFWS(Run='141195:141200', CalibrationRun='140645:140650', CalibrationOption='Interpolate')
```
Plot the spectrum 0 of the 2 outputs, they should be parallel curves roughly within 0.01 distance.
The files for this check are attached. Fine for probing the fix, they are not perfect as a reference to build a system test upon, since in this example it is actually extrapolation that is performed. Hence another test will be added in a separate PR with better reference data.

[test-files.zip](https://github.com/mantidproject/mantid/files/1102516/test-files.zip)

Fixes #19844 .



---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
